Created search bar

hamdielzard opened the pull request.

hamdielzard committed:
Added search bar

AaronSalo reviewed (commented):
Good but i dont think its ready to be pushed to develop? Still need category filter search function before pushing to develop, maybe still blocked from what graham is doing

Good progress.

AaronSalo committed:
Added a filterByCategory function

Added a function that can filter by a specified category. Currently does
not dynamically update the list, but kind of starting on that.

AaronSalo committed:
filterByCategory and filterByShoppingList done

Can now filter by both category and shopping list items. Template
buttons just at the bottom of the screen for now
